runtime/dart_isolate.h

```diff
@@ -503,21 +468,58 @@ class DartIsolate : public UIDartState {
 
   // |Dart_IsolateShutdownCallback|
   static void DartIsolateShutdownCallback(
-      std::shared_ptr<DartIsolate>* isolate_group_data,
+      std::shared_ptr<DartIsolateGroupData>* isolate_group_data,
       std::shared_ptr<DartIsolate>* isolate_data);
 
   // |Dart_IsolateCleanupCallback|
   static void DartIsolateCleanupCallback(
-      std::shared_ptr<DartIsolate>* isolate_group_data,
+      std::shared_ptr<DartIsolateGroupData>* isolate_group_data,
       std::shared_ptr<DartIsolate>* isolate_data);
 
   // |Dart_IsolateGroupCleanupCallback|
   static void DartIsolateGroupCleanupCallback(
-      std::shared_ptr<DartIsolate>* isolate_group_data);
+      std::shared_ptr<DartIsolateGroupData>* isolate_group_data);
 
   FML_DISALLOW_COPY_AND_ASSIGN(DartIsolate);
 };
 
+using ChildIsolatePreparer = std::function<bool(DartIsolate*)>;
+
+// Object holding state associated with a Dart isolate group.  An instance of
+// this class will be provided to Dart_CreateIsolateGroup as the
+// isolate_group_data.
+class DartIsolateGroupData {
```

Comment:
Let's move this into its own translation unit and also document the threading characteristics and how it differs from just the `DartIsolateData`.

Reply:
Nit: Consider using doxygen documentation format to document the class and public members as done in DartIsolate.